gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/DefaultPubSubRpc.java

```diff
@@ -50,48 +54,78 @@
 import com.google.pubsub.v1.Subscription;
 import com.google.pubsub.v1.Topic;
 
+import io.grpc.ManagedChannel;
+import io.grpc.Status.Code;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+
 import org.joda.time.Duration;
 
 import java.io.IOException;
 import java.util.Set;
 import java.util.concurrent.Future;
-
-import autovalue.shaded.com.google.common.common.collect.Sets;
-import io.grpc.Status.Code;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
 
 public class DefaultPubSubRpc implements PubSubRpc {
 
   private final PublisherApi publisherApi;
   private final SubscriberApi subscriberApi;
+  private final ScheduledExecutorService executor =
+      MoreExecutors.getExitingScheduledExecutorService(new ScheduledThreadPoolExecutor(8));
```

Comment:
> I don't see the problem with that. If more executors are created more must be closed.

Are you sure closing the executor will *also* remove it from the Shutdown hook?

> For the cost of creating threads. The library uses threads, it does not seem a waste of resources to me to always have a minimum-size pool ready to be more responsive.

If we have one thread-pool for all gcloud-java service that would make sense. Otherwise it is a waste.
If one does not call a service for N minutes. At some point the cost balance of creating a thread and caching vs just holding the resource for that long will shift.
for too long